web-platform/website/src/services/AuthenticationService.tsx

```diff
@@ -1,38 +1,73 @@
+import { onAuthStateChanged, signOut } from "firebase/auth";
 import { createContext, useContext, useEffect, useState } from "react";
 
 import { User } from "../models/User";
+import { useFirebase } from "./FirebaseService";
+import { signupWithEmail as signupWithEmailFunction } from "./UserAccountService";
 
-export type UserDataContext = {
+export interface UserDataContext {
   isLoaded: boolean;
   user?: User;
   isLoggingIn: boolean;
+  isUserVerified: boolean;
   logout: () => Promise<void>;
-};
+  signupWithEmail(email: string, password: string): Promise<void>;
+}
 
 const AuthenticationContext = createContext<UserDataContext>({
-  isLoaded: true,
+  isLoaded: false,
   user: undefined,
   isLoggingIn: false,
-  logout: async () => {},
-});
+  isUserVerified: false,
+} as UserDataContext);
 
 export function useAuthentication() {
   return useContext(AuthenticationContext);
 }
 
 export function AuthenticationProvider(props: { children: React.ReactNode }) {
-  const [user, setUser] = useState<User>();
+  const [user, setUser] = useState<User | undefined>();
+  const [isUserVerified, setIsUserVerified] = useState(false);
   const [isLoggingIn] = useState(false);
   const [isLoaded, setIsLoaded] = useState(false);
 
+  const { auth } = useFirebase();
+
   useEffect(() => {
-    // TODO: Retrieve user here...
+    onAuthStateChanged(auth, (firebaseUser) => {
+      setUser(firebaseUser ? { firebaseUser } : undefined);
+    });
+
     setIsLoaded(true);
   }, []);
 
+  useEffect(() => {
+    setIsUserVerified(
+      Boolean(
+        user &&
+          user.firebaseUser.providerData &&
+          user.firebaseUser.providerData.length &&
+          ((user.firebaseUser.providerData[0].providerId === "password" &&
+            user.firebaseUser.emailVerified) ||
+            user.firebaseUser.providerData[0].providerId === "google")
+      )
+    );
+  }, [user]);
+
   async function logout() {
-    // TODO: Clear user here...
-    setUser(undefined);
+    signOut(auth);
```

Comment:
need `await` here I think

Reply:
We don't need to (since whenever signOut completes onAuthStateChanged is called) but I will make this regardless since we have an outer async method. Doing it next PR.